src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/Sample.java

```diff
@@ -70,4 +76,18 @@ public Long getId() {
     public AccessionVersionId getAccessionVersionId() {
         return accessionVersionId;
     }
+
+    @PrePersist
+    public void validateTaxonomiesPersist() {
+        if (this.taxonomies == null || this.taxonomies.size() == 0 || this.taxonomies.get(0) == null) {
```

Comment:
taxonomies == null is being checked in @NotNull i guess

Reply:
Sure.
@selvaebi : These operations should throw error as well , because at this point the sample doesn't have any taxonomies associated with it.
I verified `@PreRemove`, `@PreDestroy` both in Sample and Taxonomy. We don't receive callbacks for removing the association between Taxonomy and Sample. Only when Sample is updated with PUT or PATCH i could trace.

Reply:
I will search any other way or trace `@JoinTable` ...

Reply:
Event Handlers are used